lib/conjur/api/router/v5.rb

```diff
@@ -16,10 +16,13 @@ def authn_authenticate account, username
         end
 
         # For v5, the authn-local message is a JSON string with account, sub, and optional fields.
-        def authn_authenticate_local username, account, expiration, cidr, &block
+        # Optional fields include the service_id and authn_type for a custom authenticator.
+        def authn_authenticate_local username, account, expiration, cidr, service_id, authn_type, &block
```

Comment:
The changes I'm about to make in the UI are likely to need this function in the v4 router. Could you add it there, too, please?

Reply:
yeah, I'm working on that now :)

Reply:
I've added an update to the v4 router.